### PR TITLE
Added database support with modified entities.

### PR DESCRIPTION
**Pull request by youssefsoli**
Database support with firebase was added with modifications to the entities in order to work with firebase.

### CODE REVIEW

**Review by modaserMoj — APPROVED** [review]
Database implementation is appropriate.

File: src/main/java/datastore/Datastore.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class Datastore
+{
+    private static final String DATABASE_URL = "https://fitbook-746a5.us-east1.firebaseio.com/";
+    private final Firestore database;
+    private static Datastore instance;
```
Comment (JacketPants12) [inline src/main/java/datastore/Datastore.java:18]:
```suggestion
    private static volatile Datastore instance;
```
To ensure thread safety

File: src/main/java/entity/User.java [hunk 2]
```diff
@@ -6,19 +6,16 @@
...
 
 public class User {
 
     @NotEmpty
     private String id;
-    private List<@Valid User> followingList;
-    private List<@Valid User> followerList;
-    private List<@Valid Post> postList;
-    private List<@Valid Post> likedPostList;
+    private List<String> following;
```
Comment (JacketPants12) [inline src/main/java/entity/User.java:15]:
```suggestion
    private List<String> followingIdList;
```

Add id to variable name for clarity of what list stores

File: src/main/java/entity/User.java [hunk 3]
```diff
@@ -6,19 +6,16 @@
...
+    private List<String> followers;
```
Comment (JacketPants12) [inline src/main/java/entity/User.java:16]:
```suggestion
    private List<String> followerIdList;
```
Add id to variable name for clarity of what list stores

File: src/main/java/entity/User.java [hunk 4]
```diff
@@ -6,19 +6,16 @@
...
+    private List<String> postList;
```
Comment (JacketPants12) [inline src/main/java/entity/User.java:17]:
```suggestion
    private List<String> postIdList;
```
Add id to variable name for clarity of what list stores

File: src/main/java/entity/User.java [hunk 5]
```diff
@@ -6,19 +6,16 @@
...
+    private List<String> likedPostList;
```
Comment (JacketPants12) [inline src/main/java/entity/User.java:18]:
```suggestion
    private List<String> likedPostIdList;
```
Add id to variable name for clarity of what list stores

**Review by feiwun — APPROVED** [review]

**Review by Ludosss — APPROVED** [review]
very cool

**Review by JacketPants12 — CHANGES_REQUESTED** [review]
Address requested changes.

**Review by JacketPants12 — CHANGES_REQUESTED** [review]
Missing, Datastore file in changes.

File: src/main/java/entity/Post.java [hunk 6]
```diff
@@ -24,12 +26,17 @@ public class Post {
     @NotEmpty
     private List<@Valid Exercise> exerciseList;
```
Comment (JacketPants12) [inline src/main/java/entity/Post.java:27]: Do we keep Exercises as POJO or maybe change to ids as in User class for posts etc..

**Comment by youssefsoli** [comment]
Oops forgot to git add

File: src/main/java/entity/Post.java (hunk 6 above)
Comment (youssefsoli) [reply src/main/java/entity/Post.java:27]: Yeah we can use ids

**Review by JacketPants12 — APPROVED** [review]
lgtm

**Review by modaserMoj — APPROVED** [review]

**Review by jpablo2002 — APPROVED** [review]
lgtm!

**Pull request merged**